### DISCUSSION TITLE
Babashka running natively on Apple Silicon with GraalVM 22.2!

### DISCUSSION THREAD

**Discussion by shrik450** [babashka/babashka#1242] [General]
Last month, the GraalVM team started releasing [dev builds of GraalVM 22.2](https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest) for Apple SIlicon (`darwin-aarch64`). I wanted to run babashka natively, so I tried compiling bb with that version and... it works! 

<img width="525" alt="image" src="https://user-images.githubusercontent.com/13411240/163541353-9d83db4a-5c53-4c08-b719-139db94dc73a.png">

I ran the native tests after compiling and they all succeeded. The bb executable size is *92M* though, so I'm not sure if I messed something up or it genuinely is just that big. I didn't change any of the feature flags and did a clean `scripts/uberjar` then `scripts/compile`.

I wouldn't say this is ready for prime time - these are dev builds of GraalVM, after all - but if you're on an M1 and want to work without rosetta I think you can go for it!

**Comment by borkdude** [comment]
Also docs about it here: https://github.com/babashka/babashka/wiki/Compiling-for-M1
